CONTRIBUTING.md

```diff
@@ -0,0 +1,72 @@
+## Contribution Guidelines
+
+Please note that this project is released with a [Contributor Code of Conduct](CODE-OF-CONDUCT.md). By participating in this project you agree to abide by its terms.
+
+## Intro
+
+Early into the AY12/13 academic year, Prof Tay Yong Chiang organized a supper for Computer Science freshmen at Tembusu College. The bunch of seniors who were gathered there put together a document for NUS computing freshmen. This is that document.
+
+Feel free to create a pull request to edit or add to it, and share it with other freshmen you know.
+
+It is now currently maintained and updated by NUS Hackers.
+
+## What is this document?
+
+This document is a [living document](https://en.wikipedia.org/wiki/Living_document) that aims to provide valuable resources and guidance to incoming
+computer science freshmen.
+
+## The goal of this document
+
+The primary goal of this document is to create a repository of knowledge that will cover various aspects
+of a freshmen's computing journey. We hope that this document will serve to ease the transition of new students
+into the School of Computing (SoC).
+
+## Contributing to the document
+
+As much as possible, we want to adhere to the the overarching ideas in the
+[awesome manifesto](https://github.com/sindresorhus/awesome/blob/main/awesome.md).
+
+In other words:
+- It's curation, not collection. - We want to avoid the document from being too lengthy or diluted.
+- Research if the stuff you're including is actually *awesome*. - Only put stuff in that you or another contributor can **personally** recommend.
+- You should rather leave stuff out than include too much. - If you're unsure, ask someone else, or comment it in the PR.
+- Comment on why something is *awesome* (Not relevant for advice) - If you're recommending a book, or a lecture, tell us why you think it's great.
+- Pay attention to grammar. - Ensure your stuff is grammatically correct, typo-free and has no Markdown formatting errors.
+
+## I'm ready to contribute! How do I start?
+
+**Make sure your content follows the general principles in [Contributing to the document](#contributing-to-the-document)**
```

Comment:
Shall we also mention that they're free (and we encourage them) to submit a PR if they want to add anything? I know it's mentioned somewhere below but might be better to include it at the start instead. What do you think?

Reply:
Sounds good